components/camel-hdfs/src/main/java/org/apache/camel/component/hdfs/HdfsConsumer.java

```diff
@@ -220,4 +233,19 @@ private HdfsInputStream createInputStream(FileStatus fileStatus) {
         }
     }
 
+    protected void updateNewExchange(Exchange exchange, int index, HdfsInputStream hdfsFile) {
+        // do not share unit of work
+        exchange.setUnitOfWork(null);
```

Comment:
This is dangerous, lets revisit why you do this. Maybe what you want to do is to create a correlated copy of the exchange where you can avoid sharing the unit of work, as it has its own uow. See ExchangeHelper and try to git grep the source code and find other components that does this

Reply:
I ported this over from the **Splitter** class, **updateNewExchange(...)** , not saying that it's right.
Will look into other options.

Reply:
Ah the splitter should have created a correlated copy with that helper class, so its a copy of the original exchange, then setting it as null is okay.